bluesky_queueserver/manager/profile_tools.py

```diff
@@ -289,3 +290,52 @@ def load_devices_from_happi(device_names, *, namespace, **kwargs):
     ns_dict = ns.__dict__
     namespace.update(ns_dict)
     return list(ns_dict)
+
+
+_env_re_worker_active = "QSERVER_RE_WORKER_ACTIVE"
+
+
+def set_re_worker_active():
+    """
+    Set the environment variable used to determine if the current process is RE Worker process.
+    Subsequent calls to ``is_re_worker_active`` in the current process will return ``True``.
+    This function should not be used in the startup scripts.
+    """
+    os.environ[_env_re_worker_active] = ""
+
+
+def clear_re_worker_active():
+    """
+    Clear the environment variable used to determine if the current process is RE Worker process.
+    Subsequent calls to ``is_re_worker_active`` in the current process will return ``False``.
+    This function should not be used in the startup scripts.
+    """
+    if _env_re_worker_active in os.environ:
+        del os.environ[_env_re_worker_active]
+
+
+def is_re_worker_active():
+    """
+    The function can be used in startup scripts or modules to check if the script is imported or
+    executed in RE Worker environment. For example, an experimental plan may contain interactive
+    features that should be disabled if the plan is executed remotely:
+
+    .. code-block:: python
+
+        from bluesky_queueserver.manager.profile_tools import is_re_worker_active
+
+        ...
+
+        if is_re_worker_active():
+            (code without interactive features, e.g. reading data from a file)
+        else:
+            (code with interactive features, e.g. manual data input)
+
+        ...
+
+    Returns
+    -------
+    boolean
+        ``True`` - the code is executed in RE Worker environment, otherwise ``False``.
+    """
+    return _env_re_worker_active in os.environ
```

Comment:
This condition will need to be updated if the previous suggestion is implemented.

Reply:
I don't think that we are going to be setting this variable externally, but it can be changed I guess.

Reply:
I see. As it's going to be used from inside the code, maybe it's worth adding a leading underscore (such as `_QSERVER_RE_WORKER_ACTIVE`) to mark it as a "private" env var (strictly speaking, bash does not care about it, but we can follow Python's style here).

Reply:
I am not sure about changing the environment variable name. May be it is better to keep it standard.

Reply:
In principle I wouldn't exclude potential to use the command like this: `$ QSERVER_RE_WORKER_ACTIVE=1 <our command>`.